src/test/java/wolox/training/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,284 @@
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(json)
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+        System.out.println("whenAddAnUser Passed");
+    }
+
+    @Test
+    void whenAddAnUserWithUsernameAlreadyTaken_thenThrowConflict() throws Exception {
+        //Given
+        User userTest = new User(1L,"Username","David", LocalDate.parse("2004-09-25"),new ArrayList<>());
+        User userTestException = new User(2L,"Username","Francisco", LocalDate.parse("2000-08-24"),new ArrayList<>());
+
+        given(userRepository.findByUsername(userTestException.getUsername())).willReturn(java.util.Optional.of(userTest));
+
+        String jsonException = mapper.writeValueAsString(userTestException);
+
+        // When + Then
+
+        mvc.perform(MockMvcRequestBuilders.post("/api/users")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(jsonException)
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isConflict());
+        System.out.println("whenAddAnUserWithUsernameAlreadyTaken Passed");
+    }
+
+    @Test
+    void whenUpdateAnUser_thenReturnOk() throws Exception {
+        //Given
+        User userTest = new User(1L,"Username","David", LocalDate.parse("2004-09-25"),new ArrayList<>());
+
+        given(userRepository.save(userTest)).willReturn(userTest);
+
+        userTest.setName("Santiago");
+        given(userRepository.save(userTest)).willReturn(userTest);
+        given(userRepository.findById(userTest.getId())).willReturn(java.util.Optional.of(userTest));
+
+        String json = mapper.writeValueAsString(userTest);
+
+        // When + Then
+
+        mvc.perform(MockMvcRequestBuilders.put("/api/users/1")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(json)
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+        System.out.println("whenUpdateAnUser Passed");
+    }
+
+    @Test
+    void whenUpdateAnUserIdMismatch_thenThrowIdMismatched() throws Exception {
+        //Given
+        User userTest = new User(1L,"Username","David", LocalDate.parse("2004-09-25"),new ArrayList<>());
+
+        given(userRepository.save(userTest)).willReturn(userTest);
+
+        userTest.setName("Santiago");
+        given(userRepository.save(userTest)).willReturn(userTest);
+        given(userRepository.findById(userTest.getId())).willReturn(java.util.Optional.of(userTest));
+
+        String json = mapper.writeValueAsString(userTest);
+
+        // When + Then
+
+        mvc.perform(MockMvcRequestBuilders.put("/api/users/2")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(json)
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isBadRequest());
+        System.out.println("whenUpdateAnUserIdMismatch Passed");
```

Comment:
This print is not necessary, as it is a test, the result is already shown by the console